x/swaprouter/creator.go

```diff
@@ -84,14 +90,17 @@ func validateCreatePoolMsg(ctx sdk.Context, msg types.CreatePoolMsg) error {
 
 	initialPoolLiquidity := msg.InitialLiquidity()
 	numAssets := initialPoolLiquidity.Len()
-	if numAssets < types.MinPoolAssets {
-		return types.ErrTooFewPoolAssets
-	}
-	if numAssets > types.MaxPoolAssets {
-		return errors.Wrapf(
-			types.ErrTooManyPoolAssets,
-			"pool has too many PoolAssets (%d)", numAssets,
-		)
+
+	if msg.GetPoolType() != types.Concentrated {
```

Comment:
Why would we not run this for the concentrated pool? I suggest keeping this as it was. If we want the CL pool have stricted bounds, we can have an  extra check in `InitializePool`.

The swaproter logic should not be concerned with what kind of pool is being created.

That is, we should avoid `msg.GetPoolType() != types.Concentrated` in swaprouter as much as possible

Reply:
Yea, I suppose as long as we check num assets in the Validate of CL itself then this is fine, will add it back

Reply:
Actually, after I removed it I remembered why it was needed. The length of pool is checked via pool liquidity, and since we make CL pools with zero liquidity, this check wont pass. Lmk if you want me to approach this in another way, but I personally don't see a problem with adding this if statement here

Reply:
I think we might want to move this to each of `MsgCreateBalancerPool` and `MsgCreateStableswapPool`'s `ValidateBasic`.

We can still create a reusable helper that does these checks and simply call that helper from every applicable `ValidateBasic`.

Let me know what you think

Reply:
It isn't reusable since StableSwap calls Coins direct and Balancer calls a PoolAssets array. I do agree with defining directly in the validate basic and will do so now

Reply:
Oh actually they are already defined!

https://github.com/osmosis-labs/osmosis/blob/244b28a450d3e64c24d2c8a71c4fe56cf7865348/x/gamm/pool-models/stableswap/msgs.go#L63-L67

and 

https://github.com/osmosis-labs/osmosis/blob/244b28a450d3e64c24d2c8a71c4fe56cf7865348/x/gamm/pool-models/balancer/msgs.go#L42-L45

Reply:
I could actually generalize this if we move it up to the swaprouter level, since we then have commonality at the msg.InitialLiquidity level. Otherwise I will just leave the respective logic to be done at the ValidateBasic level (i think this is the best approach imo)

Reply:
I'm supportive of leaving it as is in gamm for this PR to limit the scope. However, is my understanding correct that we would still remove this from the swap router?

Reply:
Yes, otherwise we are just validating twice